source/slang/slang-check-stmt.cpp

```diff
@@ -327,6 +329,260 @@ namespace Slang
         stmt->expression = CheckExpr(stmt->expression);
     }
 
+    void SemanticsStmtVisitor::tryInferLoopMaxIterations(ForStmt* stmt)
+    {
+        DeclRef<Decl> initialVar = {};
+        DeclRef<Decl> predicateVar = {};
+        Expr* initialVal = nullptr;
+        if (auto varStmt = as<DeclStmt>(stmt->initialStatement))
+        {
+            auto varDecl = as<VarDecl>(varStmt->decl);
+            if (!varDecl)
+                return;
+            initialVar.decl = varDecl;
+            initialVal = varDecl->initExpr;
+        }
+        else if (auto exprStmt = as<ExpressionStmt>(stmt->initialStatement))
+        {
+            auto assignExpr = as<AssignExpr>(exprStmt->expression);
+            if (!assignExpr)
+                return;
+            auto varExpr = as<VarExpr>(assignExpr->left);
+            if (!varExpr)
+                return;
+            initialVar = varExpr->declRef;
+            initialVal = assignExpr->right;
+        }
+        else
+            return;
+
+        auto initialLitVal =
+            as<ConstantIntVal>(tryFoldIntegerConstantExpression(initialVal, nullptr));
+
+
+        ConstantIntVal* finalVal = 0;
+        auto binaryExpr = as<InfixExpr>(stmt->predicateExpression);
+        if (!binaryExpr)
+            return;
+        auto compareFuncExpr = as<DeclRefExpr>(binaryExpr->functionExpr);
+        if (!compareFuncExpr)
+            return;
+        if (!compareFuncExpr->declRef.getDecl())
+            return;
+        IROp compareOp = kIROp_Nop;
+        if (auto intrinsicOpModifier = compareFuncExpr->declRef.getDecl()->findModifier<IntrinsicOpModifier>())
+        {
+            compareOp = (IROp)intrinsicOpModifier->op;
+        }
+        else
+        {
+            return;
+        }
+        if (binaryExpr->arguments.getCount() != 2)
+            return;
+        auto leftCompareOperand = binaryExpr->arguments[0];
+        auto rightCompareOperand = binaryExpr->arguments[1];
+        while (auto typeCast = as<TypeCastExpr>(leftCompareOperand))
```

Comment:
Stripping the cast/s seems as if it could be a little dangerous? Say in a rather contrived example
```
for (int i = 0x101; uint8_t(i) < uint8_t(0xffffff); ++i) { ... }
```

Reply:
You are right, removed.